File: 150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        int n = tokens.size();
        
        for (int i = 0; i < n; i++) {
            string token = tokens[i];
            if (token == "/" || token == "+" || token == "-" || token == "*") {
                int top = st.top(); st.pop();
                int secondTop = st.top(); st.pop();
                if (token == "/") {
                    secondTop /= top;
                } else if (token == "*") {
                    secondTop *= top;
                } else if (token == "+") {
                    secondTop += top;
                } else if (token == "-") {
                    secondTop -= top;
                }
                st.push(secondTop);
            } else {
                st.push(stoi(token));
            }
        }
        
        return st.top();
    }
};

File: 150. Evaluate Reverse Polish Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int n = tokens.length;
        for(int i = 0; i < n; i++){
            if(tokens[i].equals("/") || tokens[i].equals("+") || tokens[i].equals("-") || tokens[i].equals("*")){
                int top = stack.pop();
                int secondTop = stack.pop();
                switch(tokens[i]){
                    case "/":
                        secondTop /= top;
                        break;
                    case "*":
                        secondTop *= top;
                        break;
                    case "+":
                        secondTop += top;
                        break;
                    case "-":
                        secondTop -= top;
                        break;
                }
                stack.push(secondTop);
            }
            else{
                stack.push(Integer.parseInt(tokens[i]));
            }
        }

        return stack.pop();
    }
}